types/lodash/scripts/generate-fp.ts

```diff
@@ -127,24 +127,6 @@ async function main() {
         if (err)
             console.error("Failed to write fp.d.ts: ", err);
     });
-
-    // Make sure the generated files are listed in tsconfig.json, so they are included in the lint checks
-    const tsconfigPath = path.join("..", "tsconfig.json");
```

Comment:
is the tsconfig generated somewhere else now or is it not needed?

Reply:
> is the tsconfig generated somewhere else

Yes. Here: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/bf8afd29658567c0c5cd447b3e508a9ec9dcaaf3/types/lodash/scripts/generate-modules.ts#L21

---

Apart from that, I was not sure if it's really ok to remove the here, but thought there would be no drawbacks in practice and decided to do so.
The details are below:

What the code here does is to add `fp/*` files to the "files" section in `tsconfig.json`, and if the code remains, this script `generate-fp.ts` overwrites `tsconfig.json` with the content like the following
```json
{
    "compilerOptions": {
        "module": "commonjs",
        "lib": [
            "es6"
        ],
        "noImplicitAny": true,
        "noImplicitThis": true,
        "strictNullChecks": true,
        "strictFunctionTypes": true,
        "baseUrl": "../",
        "typeRoots": [
            "../"
        ],
        "types": [],
        "noEmit": true,
        "forceConsistentCasingInFileNames": true
    },
    "files": [
        "index.d.ts",
        "lodash-tests.ts",
        "fp/add.d.ts",
        "fp/after.d.ts",
...(many lines)...
        "fp/placeholder.d.ts"
    ]
}
```

However, `tsconfig.json` actually committed to this repo does not include these lines ( https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/lodash/tsconfig.json#L20-L23 ),
which means the authors have never used `generate-fp.ts` for a long time or they have ignored the generated `tsconfig.json` to commit.
And when I removed the code here like this PR, the generated content of `tsconfig.ts` exactly matches the current committed one - this modified `generate-fp.ts` outputs `tsconfig.json` as-is.
That's why I thought there would be no drawbacks *in practice* if I delete the code here and decided to do so.

Reply:
Looks like those files were removed from tsconfig relatively recently (Nov 2019, #40676) so it is correct to remove this code. Its purpose was to add all of the "fp/" files but it looks like that is no longer necessary.

However, it looks like `index.d.ts` does not reference `fp.d.ts`, so I wonder if we should add `fp.d.ts` to the files list.